build/image_checks/checkDuplicatedImages.js

```diff
@@ -0,0 +1,62 @@
+const path      = require('path');
+const common    = require('../../scripts/common');
+const {
+    getDefaultPath,
+    gitIgnoreToArray,
+    fetchFiles,
+    flatten } = require('../utils/cli');
+
+const ignored_files = gitIgnoreToArray();
+
+const ignored_extensions = ['DS_Store'];
+
+module.exports = function(grunt) {
+    grunt.initConfig({
+        checkDuplicatedImages: {},
+    });
+
+    grunt.registerTask('check_duplicated_images', 'Check for duplicate images based on their content. ', () => {
+        const default_path = getDefaultPath(grunt.option('section'));
+        const where_to_look = path.resolve(common.root_path, `src/images/${default_path}/`);
+        grunt.log.writeln(`Checks for duplicate images based on their content.\nSearching: src/images/${default_path}`);
+
+        // Actual test is running from here. We can check for other directories as well, but it is highly unlikely that
+        // someone put a duplicate in other folders. This grunt task will accept a --all option. --all option will try
+        // to check every single file inside images directory. if no option is provided, only the images inside
+        // app_2 will be processed.
+        const flattened = flatten(fetchFiles(where_to_look))
+            .filter(item => !!item &&
```

Comment:
Would it be possible to add filter utils and structure the filter to something like this?
Pros: can reuse the filtering logic here and in `checkImagesNamingConvention.js` and `checkOrphanedImages.js`

```javascript
.filter(Boolean) // same as filter(item => !!item)
.filter(isNotIgnoredFileCommonRoot)
.filter(isNotIgnoredExtension)

// filter utils:
const isNotIgnoredFileCommonRoot = (item) => !ignored_files.some(ignored => path.resolve(common.root_path, ignored) === item.filename);

const isNotIgnoredFile = (item) => !ignored_files.some(ignored => ignored === item.filename);

const isNotIgnoredExtension = (item) => {
  return !ignored_extensions.some(ignored => {
        const matchingGroups = item.filename.match(/\.[\w\d]{1,5}$/i);
        return matchingGroups && matchingGroups[0].substring(1) === ignored;
  }));
}
```

Reply:
It was like this at the beginning, but decided to change the structure to make it more fast and run the filters only once.